### PR TITLE
LogMaskedAttribute to handle int,long and guid values

### PR DESCRIPTION
**Pull request by andreas-moneygate**

### CODE REVIEW

**Comment by sungam3r** [comment]
Please see https://github.com/destructurama/attributed/pull/122#issuecomment-2526204704 . What do you think?

**Comment by andreas-moneygate** [comment]
> Please see [#122 (comment)](https://github.com/destructurama/attributed/pull/122#issuecomment-2526204704) . What do you think?

How about having a property in LogMaskedAttribute allowing other types besides string to be masked?

**Comment by sungam3r** [comment]
I thought again about it. All non-string properties marked with this attribute ended in default case with `ScalarValue.Null`. So my reasoning from https://github.com/destructurama/attributed/pull/122#issuecomment-2526204704 is wrong. If now anyone marks int property with that attribute then it effectively becames `null`.

**Comment by sungam3r** [comment]
I added `LogMaskedAttribute_Nullify_Bool_Property` test to verify. Please resolve conflicts and I'll review again.

**Comment by sungam3r** [comment]
Also you may include changes from #122.

File: src/Destructurama.Attributed.Tests/MaskedAttributeTests.cs [hunk 1]
```diff
@@ -690,4 +714,82 @@
...
+        {
+            DefaultMaskedLong = long.MaxValue
+        };
+
+        var evt = DelegatingSink.Execute(customized);
+
+        var sv = (StructureValue)evt.Properties["Customized"];
+        var props = sv.Properties.ToDictionary(p => p.Name, p => p.Value);
+
+        props.ContainsKey("DefaultMaskedLong").ShouldBeTrue();
```
[the remark this answers is not in the document]
Comment (andreas-moneygate) [reply src/Destructurama.Attributed.Tests/MaskedAttributeTests.cs:758]: @sungam3r  Shall I resolve this with something like this?

```
 var sv = (StructureValue)evt.Properties["Customized"];
 var prop = sv.Properties.SingleOrDefault(p => p.Name == "DefaultMaskedLong");

 prop.ShouldNotBeNull();
 prop.Value.LiteralValue().ShouldBe("***");
```

**Comment by andreas-moneygate** [comment]
@sungam3r
Conflicts resolved

File: src/Destructurama.Attributed.Tests/MaskedAttributeTests.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (sungam3r) [reply src/Destructurama.Attributed.Tests/MaskedAttributeTests.cs:758]: No. I dismiss such warnings. Not applicable.

File: README.md [hunk 2]
```diff
@@ -211,6 +223,18 @@ public class CustomizedMaskedLogs
...
+    /// 9223372036854775807 results in "922***807"
+    /// </summary>
+    [LogMasked(ShowFirst = 3, ShowLast = 3)]
+    public long? ShowFirstAndLastThreeAndDefaultMaskLongInTheMiddle { get; set; }
+
+    /// <summary>
+    /// 2147483647 results in "214****647"
+    /// </summary>
+    [LogMasked(ShowFirst = 3, ShowLast = 3, PreserveLength = true)]
+    public long? ShowFirstAndLastThreeAndDefaultMaskIntInTheMiddlePreservedLength { get; set; }
```
Comment (sungam3r) [inline README.md:236]:
```suggestion
    public int? ShowFirstAndLastThreeAndDefaultMaskIntInTheMiddlePreservedLength { get; set; }
```

File: README.md [hunk 3]
```diff
@@ -230,13 +254,19 @@ public class CustomizedMaskedLogs
     public string? ShowLastThreeThenDefaultMaskedPreservedLength { get; set; }
 
     /// <summary>
-    /// 123456789 results in "123REMOVED"
+    ///  123456789 results in "123_REMOVED_"
```
Comment (sungam3r) [inline README.md:257]:
```suggestion
    /// 123456789 results in "123_REMOVED_"
```

File: README.md [hunk 4]
```diff
@@ -230,13 +254,19 @@ public class CustomizedMaskedLogs
...
 
     /// <summary>
-    /// 123456789 results in "123_REMOVED_"
+    /// d3c4a1f2-3b4e-4f5a-9b6c-7d8e9f0a1b2c results in "d3c4a_REMOVED_"
+    /// </summary>
+    [LogMasked(Text = "_REMOVED_", ShowFirst = 5)]
+    public Guid? ShowFirstFiveThenCustomMaskGuid { get; set; }
+
+    /// <summary>
+    ///  123456789 results in "123_REMOVED_"
```
Comment (sungam3r) [inline README.md:269]:
```suggestion
    /// 123456789 results in "123_REMOVED_"
```

File: src/Destructurama.Attributed.Tests/MaskedAttributeTests.cs [hunk 5]
```diff
@@ -57,6 +69,18 @@ public class CustomizedMaskedLogs
...
+    /// 9223372036854775807 results in "922***807"
+    /// </summary>
+    [LogMasked(ShowFirst = 3, ShowLast = 3)]
+    public long? ShowFirstAndLastThreeAndDefaultMaskLongInTheMiddle { get; set; }
+
+    /// <summary>
+    /// 2147483647 results in "214****647"
+    /// </summary>
+    [LogMasked(ShowFirst = 3, ShowLast = 3, PreserveLength = true)]
+    public long? ShowFirstAndLastThreeAndDefaultMaskIntInTheMiddlePreservedLength { get; set; }
```
Comment (sungam3r) [inline src/Destructurama.Attributed.Tests/MaskedAttributeTests.cs:82]:
```suggestion
    public int? ShowFirstAndLastThreeAndDefaultMaskIntInTheMiddlePreservedLength { get; set; }
```

File: src/Destructurama.Attributed.Tests/MaskedAttributeTests.cs [hunk 6]
```diff
@@ -76,13 +100,19 @@ public class CustomizedMaskedLogs
     public string? ShowLastThreeThenDefaultMaskedPreservedLength { get; set; }
 
     /// <summary>
-    /// 123456789 results in "123REMOVED"
+    ///  123456789 results in "123_REMOVED_"
```
Comment (sungam3r) [inline src/Destructurama.Attributed.Tests/MaskedAttributeTests.cs:103]:
```suggestion
    /// 123456789 results in "123_REMOVED_"
```

File: src/Destructurama.Attributed.Tests/MaskedAttributeTests.cs [hunk 7]
```diff
@@ -76,13 +100,19 @@ public class CustomizedMaskedLogs
...
 
     /// <summary>
-    /// 123456789 results in "123_REMOVED_"
+    /// d3c4a1f2-3b4e-4f5a-9b6c-7d8e9f0a1b2c results in "d3c4a_REMOVED_"
+    /// </summary>
+    [LogMasked(Text = "_REMOVED_", ShowFirst = 5)]
+    public Guid? ShowFirstFiveThenCustomMaskGuid { get; set; }
+
+    /// <summary>
+    ///  123456789 results in "123_REMOVED_"
```
Comment (sungam3r) [inline src/Destructurama.Attributed.Tests/MaskedAttributeTests.cs:115]:
```suggestion
    /// 123456789 results in "123_REMOVED_"
```

**Comment by mickeyperlstein** [comment]
Although GUID's are not strings, we treat them as such, and do sometimes put in them secure data, not PII but machine II for sure.

**Comment by mickeyperlstein** [comment]
I also did not appreciate that it took you so  long to merge my solution, and that you changed it so it's not my solution is even less appreciative.

It was MY idea, it was MY solution - and yet I get no credit in your product, last time im adding source to your code

**Comment by sungam3r** [comment]
https://github.com/destructurama/attributed/releases/tag/4.1.0

Thanks one more time, @mickeyperlstein

**Comment by sungam3r** [comment]
Sorry, it was not intentional, I included you in release notes.

**Comment by sungam3r** [comment]
With #141 you can mask property of any type, no more need to add custom cases.

**Pull request merged**